addons/mass_mailing/wizard/mailing_mailing_test.py

```diff
@@ -48,10 +48,27 @@ def send_mail_test(self):
                 'notification': True,
                 'mailing_id': mailing.id,
                 'attachment_ids': [(4, attachment.id) for attachment in mailing.attachment_ids],
-                'auto_delete': True,
                 'mail_server_id': mailing.mail_server_id.id,
             }
             mail = self.env['mail.mail'].sudo().create(mail_values)
             mails_sudo |= mail
         mails_sudo.send()
+
+        notification_messages = []
+        for mail_sudo in mails_sudo:
+            if mail_sudo.state == 'sent':
+                notification_messages.append(
+                    _('<li>Test e-mail successfully sent to %s</li>', mail_sudo.email_to))
+                # manually delete the email if it was successfully sent
+                mail_sudo.unlink()
+            elif mail_sudo.state == 'exception':
```

Comment:
Not sure we should keep it. Either you have an easy way to send it again (which is unsure), either user may simply test again.

Reply:
The goal is to give some feedback to the user so that he can know it failed.

Otherwise he presses "Send", nothing happens in the interface, and he doesn't receive anything in its mailbox.
(Also, I think that if we notify a success we should also notify a failure).

I agree it would be best if he could easily retry with a button but it would be complicated to implement and as you said he can just press "Test" again.

Reply:
I mean, why not preparing the notification message then unlinking all mails ? I don't think we should keep failed mails, especially people do not necessarily know those are available in technical/emails .